generators/parser.go

```diff
@@ -0,0 +1,58 @@
+package generators
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type KindGroup struct {
```

Comment:
Feel free to unexport these structs for the timebeing - we'll likely replace these in the next iteration.